fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -2041,29 +2041,31 @@ private void recalculateLoanCharge(final LoanCharge loanCharge, final int penalt
         BigDecimal amount = BigDecimal.ZERO;
         BigDecimal chargeAmt;
         BigDecimal totalChargeAmt = BigDecimal.ZERO;
-        if (loanCharge.getChargeCalculation().isPercentageBased()) {
-            if (loanCharge.isOverdueInstallmentCharge()) {
-                amount = calculateOverdueAmountPercentageAppliedTo(loanCharge, penaltyWaitPeriod);
+        if (!loanCharge.isOverdueInstallmentCharge()) {
+            if (loanCharge.getChargeCalculation().isPercentageBased()) {
+                if (loanCharge.isOverdueInstallmentCharge()) {
```

Comment:
@fiter-julius-oketayot This statement will never be true because on line 2044 we are checking the opposite of it